samples/calculator/Adder/OperationsImpl.cs

```diff
@@ -1,14 +1,130 @@
+            var safeOwner = owner.Trim();
+            var safeName = name.Trim();
+            // Very basic validation to avoid path injection
+            if (safeOwner.Contains('/') || safeName.Contains('/'))
+                return "{\"error\":\"invalid-slash\"}";
+            var url = $"https://api.github.com/repos/{safeOwner}/{safeName}";
+
+            // Empty headers collection (GitHub typically requires a User-Agent, but per request no headers are passed).
+            // Fields headers;
+            // try
+            // {
+            //     headers = Fields.FromList(new System.Collections.Generic.List<(string, byte[])>());
+            // }
+            // catch (WitException<HeaderError> hex)
+            // {
+            //     return $"{{\"error\":\"header-build\",\"detail\":\"{hex.TypedValue.Tag}\"}}";
+            // }
+
+            var headers = new Fields();
+            try {
+                // GitHub API recommends providing a User-Agent; also add Accept for clarity.
+                headers.Set("User-Agent", new System.Collections.Generic.List<byte[]> { System.Text.Encoding.UTF8.GetBytes("componentize-dotnet-sample/1.0") });
+                headers.Set("Accept", new System.Collections.Generic.List<byte[]> { System.Text.Encoding.UTF8.GetBytes("application/vnd.github+json") });
+            } catch (WitException<HeaderError>) { /* non-fatal if immutable/forbidden */ }
+            var req = new OutgoingRequest(headers);
+            req.SetMethod(Method.Get());
+
+            var uri = new Uri(url);
+            req.SetScheme(uri.Scheme.Equals("https", StringComparison.OrdinalIgnoreCase) ? Scheme.Https() : Scheme.Http());
+            req.SetAuthority(uri.Authority);
+            req.SetPathWithQuery(uri.PathAndQuery);
+
+            RequestOptions? options = null;
+            var future = OutgoingHandlerInterop.Handle(req, options);
+
+            IncomingResponse? incoming = null;
+            const int timeoutMs = 5000; // total budget
+            var sw = System.Diagnostics.Stopwatch.StartNew();
+            int delay = 10; // start with 10ms, exponential backoff up to 250ms
+            while (sw.ElapsedMilliseconds < timeoutMs)
+            {
+                var resultOpt = future.Get();
+                if (resultOpt != null)
+                {
+                    var outer = resultOpt.Value;
+                    if (outer.IsOk)
+                    {
+                        var inner = outer.AsOk;
+                        if (inner.IsOk)
+                        {
+                            incoming = inner.AsOk;
+                        }
+                        else
+                        {
+                            var err = inner.AsErr;
+                            return $"{{\"error\":\"http-transport\",\"code\":\"{err.Tag}\"}}";
+                        }
+                    }
+                    break;
+                }
+                System.Threading.Thread.Sleep(delay);
```

Comment:
You may need to poll the response (`future.Subscribe`) instead of simply waiting.

Reply:
@jsturtevant posted this sample in #98  that uses `PollWasiEventLoopUntilResolved`. It also appears that HttpClient API can be used. That would be amazing. Going to try out https://github.com/jsturtevant/csharp-components-demos/blob/main/console/Program.cs